### PR TITLE
[JENKINS-42841] - Parent POM update and slave=>agent renaming

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Slave to agent renaming in Javadoc (No CLASSNAME changes)
- [x] Update for the Parent POM
- [x] FindBugs cleanup
- [x] License link fix in POM

https://issues.jenkins-ci.org/browse/JENKINS-42841

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,36 +3,25 @@
...
-        <extensions>true</extensions>
-      </plugin>
-    </plugins>
-  </build>
-
   <licenses>
     <license>
       <name>MIT License</name>
-      <url>http://jenkins-ci.org/mit-license</url>
+      <url>https://jenkins.io/license/</url>
```
Comment (oleg-nenashev) [inline pom.xml:24]: I would prefer a link to a standard MIT license, but I'm not sure if it is a valid change. IANAL

File: src/main/java/org/jenkinsci/modules/slave_installer/InstallationException.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package org.jenkinsci.modules.slave_installer;
 
 /**
- * Represents an anticipated failure during {@link SlaveInstaller#install(LaunchConfiguration)}
+ * Represents an anticipated failure during {@link SlaveInstaller#install(org.jenkinsci.modules.slave_installer.LaunchConfiguration, org.jenkinsci.modules.slave_installer.Prompter)}
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/modules/slave_installer/InstallationException.java:4]: Likely there was a compatibility change long-long ago. But I doubt we care now

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

**Comment by oleg-nenashev** [comment]
@reviewbybees done (review timeout). Going forward as a maintainer

**Pull request merged**